### PR TITLE
AWS API raises an exception if both AZ and VPCZoneIdentifier params a…

### PR DESCRIPTION
**Pull request by captainkerk**
In trying to fix https://github.com/spulec/moto/issues/1288, I realized that the AWS API requires either `AvailabilityZones` or `VPCZoneIdentifier`.

I think the mock_autoscaling service should also have this requirement, as boto calls without either fail.

### CODE REVIEW

File: tests/test_autoscaling/test_autoscaling.py [hunk 1]
```diff
@@ -105,7 +111,7 @@ def test_create_autoscaling_groups_defaults():
     # Defaults
     list(group.availability_zones).should.equal([])
     group.desired_capacity.should.equal(2)
-    group.vpc_zone_identifier.should.equal('')
+    group.vpc_zone_identifier.should.equal(mocked_networking['subnet1'])
```
Comment (JackDanger) [inline tests/test_autoscaling/test_autoscaling.py]: Nice!

**Comment by captainkerk** [comment]
Oops. Tests were passing locally, I'll investigate the travis failures.

**Comment by captainkerk** [comment]
Ok, the test suite is passing again. Only a few tests of the server suite were failing, the same tests using the running with server=false were passing.

In these last two commits, I created a `setup_networking_deprecated()` method to do the same thing as setup_networking, but use the older boto calls. I didn't think it was a problem with the way I was wrapping the `setup_networking()` with a mock_ec2, but when the test calling `setup_networking()` was wrapped in a mock_autoscaling_deprecated decorator, I found that the boto3 calls contained in `setup_networking()` actually went out to the AWS API. It seems like 'deprecated' and 'not' decorators cannot be intermingled? I.e. If you use the deprecated variant first, the other kind is ignored?

Is that expected?

**Comment by captainkerk** [comment]
Heyy @JackDanger sorry to be a bother, anything I could do to help move this along? 

I know I'm touching pretty much every test, but it seemed unavoidable.

**Comment by spulec** [comment]
Thanks!

**Pull request merged**